ethereum-consensus/src/deneb/blob_sidecar.rs

```diff
@@ -1,5 +1,6 @@
 use crate::{
     deneb::{
+        mainnet::BeaconBlockBody,
```

Comment:
you want to use the generic type here, not a particular preset

Reply:
i'm not sure how to import a generic BeaconBlockBody

Reply:
```suggestion
        beacon_block::BeaconBlockBody,
```

you'll need to provide the const generic arguments which is why I was suggesting just having this be a free-standing function, rather than having types all over the place

Reply:
I'm not sure how to provide the const generic arguments.  When i add the const generics to the appropriate function signatures, i get this error:

```    Checking ethereum-consensus v0.1.1 (/home/echo/Desktop/Ethereum/ethereum-consensus/ethereum-consensus)
error[E0282]: type annotations needed
  --> ethereum-consensus/src/deneb/blob_sidecar.rs:81:5
   |
81 |     BeaconBlockBody::generalized_index(path)
   |     ^^^^^^^^^^^^^^^ cannot infer the value of const parameter `MAX_PROPOSER_SLASHINGS` declared on the struct `BeaconBlockBody`
```